sympy/calculus/range_func.py

```diff
@@ -0,0 +1,170 @@
+from sympy import sympify, diff, limit, S, oo
+from sympy.core import Expr
+from sympy.core.numbers import Number
+from sympy.calculus.singularities import singularities
+from sympy.sets.sets import Interval, Intersection, FiniteSet, Union, Complement, Set
+from sympy.simplify import simplify
+from sympy.solvers.solveset import solveset_real
+
+
+def range_func(func, set_value):
```

Comment:
`codomain`[#1] is a more precise term for the function you are trying to implement here, plus it avoids the confusion with the builtin `range` function.

[#1]: https://en.wikipedia.org/wiki/Codomain